Dotnet.Homeworks.Tests/CqrsValidation/PipelineBehaviorTests.cs

```diff
@@ -0,0 +1,137 @@
+    public async Task Mediator_Should_InvokeBehaviors_And_ReturnSucceedResult_WhenCallGetAllUsers()
+    {
+        // Assert
+        await using var testEnvBuilder = new CqrsEnvironmentBuilder().WithHandlersInDi().WithPipelineBehaviors();
+        
+        testEnvBuilder.SetupHttpContextClaims(new List<Claim>()
+            { new Claim(ClaimTypes.Role, Roles.Admin.ToString()) });
+        
+        var env = testEnvBuilder.Build();
+        
+        // Act
+        var result =
+            await CqrsEnvironment.HandleQuery<GetAllUsersQuery, GetAllUsersDto>(env.GetAllUsersQueryHandler,
+                TestUsers.GetAllUsersQuery());
+        
+        // Assert
+        Assert.True(result.IsSuccess);
+    }
+    
+    [Homework(RunLogic.Homeworks.CqrsValidatorsDecorators)]
+    public async Task Mediator_Should_InvokeBehaviors_And_ReturnFailedResult_WhenCallGetAllUsersWithNoPermission()
+    {
+        // Assert
+        await using var testEnvBuilder = new CqrsEnvironmentBuilder().WithHandlersInDi().WithPipelineBehaviors();
+        testEnvBuilder.SetupHttpContextClaims(new List<Claim>());
+        
+        var env = testEnvBuilder.Build();
+        
+        // Act
+        var result =
+            await CqrsEnvironment.HandleQuery<GetAllUsersQuery, GetAllUsersDto>(env.GetAllUsersQueryHandler,
+                TestUsers.GetAllUsersQuery());
+        
+        // Assert
+        Assert.True(result.IsFailure);
+    }
+    
+    [Homework(RunLogic.Homeworks.CqrsValidatorsDecorators)]
+    public async Task Mediator_Should_InvokeBehaviors_And_ReturnFailedResult_WhenCallDeleteUserWithNoPermission()
+    {
+        // Assert
+        var email = "correct@email.ru";
+        var name = "name";
+        await using var testEnvBuilder = new CqrsEnvironmentBuilder().WithHandlersInDi().WithPipelineBehaviors();
+        var guid = await testEnvBuilder.UserRepositoryMock.Object?.InsertUserAsync(new User() {Name = name, Email = email})!;
+        testEnvBuilder.SetupHttpContextClaims(new List<Claim>());
+
+        var env = testEnvBuilder.Build();
+        
+        // Act
+        var result =
+            await CqrsEnvironment.HandleCommand(env.DeleteUserByAdminCommandHandler,
+                TestUsers.DeleteUserByAdminCommand(guid));
+        
+        // Assert
+        Assert.True(result.IsFailure);
+        env.UnitOfWorkMock.Verify(x=>x.SaveChangesAsync(It.IsAny<CancellationToken>()), Times.Never);
+    }
+    
+    [Homework(RunLogic.Homeworks.CqrsValidatorsDecorators)]
+    public async Task Mediator_Should_InvokeBehaviors_And_ReturnSucceedResult_WhenCallDeleteUser()
+    {
+        // Assert
+        var email = "correct@email.ru";
+        var name = "name";
+        await using var testEnvBuilder = new CqrsEnvironmentBuilder().WithHandlersInDi().WithPipelineBehaviors();
+        var guid = await testEnvBuilder.UserRepositoryMock.Object?.InsertUserAsync(new User() {Name = name, Email = email})!;
```

Comment:
зачем тут проверка на null и затем мы говорим, что гуид есть 100%  
возможно имеет смысл написать так
```suggestion
        var guid = await testEnvBuilder.UserRepositoryMock.Object.InsertUserAsync(new User() {Name = name, Email = email});
```  
и затем добавить проверку на null.  
Если коммент правдивый, то в других местах тоже надо перепроверить

Reply:
Убрал